components/org.wso2.carbon.identity.data.publisher.audit.user.operation/src/main/java/org/wso2/carbon/identity/data/publisher/audit/user/operation/model/UserData.java

```diff
@@ -147,7 +134,8 @@ public void addParameter(T1 key, T2 value) {
     public void addParameters(Map<T1, T2> parameters) {
         for (Map.Entry<T1, T2> parameter : parameters.entrySet()) {
             if (this.parameters.containsKey(parameter.getKey())) {
-                throw IdentityRuntimeException.error("Parameters map trying to override existing key " + parameter.getKey());
+                throw IdentityRuntimeException.error("Parameters map trying to override existing key "
+                        + parameter.getKey());
```

Comment:
What's the rationale behind restricting overriding the entry throwing a runtime exception

Reply:
These parameter bags were simply added because this is done so in the other data publishers. These parameter bags would be useful if we are going to send more metadata with events later.

Simply to make this more consistent I have complied with how the other data publishers are implemented. (One data publisher alone not restricting this would make the IS analytics to show inconsistent behavior and confuse the user.)

Reply:
How can this cause such inconsistent behaviour in analytics. This will only replace the value against the same key if such key already exists. There won't be any duplicated keys

Reply:
Not in analytics. Between data publishers.

If it's not an issue I can change it according to your suggestion. It won't anyway have any effect on the analytics done as you mentioned.

Reply:
So, this is not just about adhering to code consistency being followed in publisher implementations then.
Will accept based on that.